service/test: Fix waiting in testSessionAffinity and regroup affinity match map updates

brb opened the pull request:
See commit msgs.

Fix #11296
Partially_fix #11442

brb committed:
test: Fix waiting in testSessionAffinity

Previously, the waiting for the cilium endpoint was broken in
testSessionAffinity helper: jq didn't return error after the endpoint
has been removed, which lead to the long duration times.

Fixing the jq issue revealed, that it's not enough to wait until
the endpoint has disappeared from "cilium endpoints list".

Further debugging has shown, that the first request from outside after
the pod removal was failing because the IPCache entry was missing
for the new pod at the time the request was sent which lead to
the packet drop (see the comment in the test helper for more details).

Signed-off-by: Martynas Pumputis <m@lambda.lt>

joestringer commented on test/k8sT/Services.go:
```diff
@@ -832,14 +832,16 @@ var _ = Describe("K8sServicesTest", func() {
+			// should give enough time for cilium-agents to remove the deleted pod
+			// from the BPF LB maps, so that the next request won't choose the deleted pod.
+			waitPodsDs()
+			time.Sleep(1 * time.Second)
```
Do you think that waiting 1 second is reliable?

I've often seen `kubectl delete` commands hang for some time, more than one second. I don't have a direct comparison with this as I didn't dig into what `waitPodsDs()` does here though.

brb replied:
Apparently not. Did more troubleshooting and found out that ipcache entry is missing for the new backend after the removal which resulted in the request being dropped.

brb committed:
service: Regroup session affinity match updates

Add matches for new backends after the backends have been inserted into
the BPF maps. Otherwise, if an affinity for the svc existed before and
its backend ID == the new backend ID, then the new backend will be
selected which is missing for a short period of time in the map (until
it's inserted).

Signed-off-by: Martynas Pumputis <m@lambda.lt>

qmonnet reviewed (commented):
[Ignore this comment - I forgot to add inline comment before submitting the review, sorry for the noise]

qmonnet commented on test/k8sT/Services.go:
```diff
@@ -830,14 +830,28 @@ var _ = Describe("K8sServicesTest", func() {
+			// due to failing fib_lookup).
+			waitPodsDs()
+			if fromOutside && vxlan {
+				time.Sleep(5 * time.Second)
```
Would it make sense to create a new helper to wait for the specific ipcache entry, so that we are sure that it has been added before going on? Maybe overkill, I'm not strongly opposed to the 5s wait.

brb replied:
It's a bit an overkill, but on the other hand we are risking to introduce flakes with sleep. Going to add it in this PR.

qmonnet reviewed (commented):
Looks good, and nice catch! One question.

qmonnet reviewed (approved):
Thanks for the change, looks all good to me now.
